src/transport/SecureSessionTable.cpp

```diff
@@ -183,6 +239,109 @@ SecureSession * SecureSessionTable::EvictAndAllocate(uint16_t localSessionId, Se
 
 void SecureSessionTable::DefaultEvictionPolicy(EvictionPolicyContext & evictionContext)
 {
+    //
+    // This implements a spec-compliant sorting policy that ensures both guarantees for sessions per-fabric as
+    // mandated by the spec as well as fairness in terms of selecting the most appropriate session to evict
+    // based on multiple criteria.
+    //
+    // See the description of this function in the header for more details on each sorting key below.
+    //
+    evictionContext.Sort([&evictionContext](const SortableSession & a, const SortableSession & b) {
+        //
+        // Sorting on Key1
+        //
+        if (a.mNumMatchingOnFabric != b.mNumMatchingOnFabric)
+        {
+            return a.mNumMatchingOnFabric > b.mNumMatchingOnFabric;
+        }
+
+        if (a.mNumMatchingOnFabric == b.mNumMatchingOnFabric)
```

Comment:
Has to be true, no?  If they were not equal we aready returned.  So can remove this test and outdent...